datafusion-examples/examples/parquet_sql_multiple_files.rs

```diff
@@ -60,5 +63,45 @@ async fn main() -> Result<()> {
     // print the results
     df.show().await?;
 
+    // Second example were we temporarily move into the test data's parent directory and
+    // simulate a relative path, this requires registering an ObjectStore.
+    let cur_dir = std::env::current_dir()?;
+
+    let test_data_path = Path::new(&test_data);
+    let test_data_path_parent = test_data_path.parent().ok_or("No parent")?;
```

Comment:
.expect("No parent") since it'll fail further down with test_data_path_parent set to 'No parent'

Reply:
E.g. testing `.expect` (top) vs `?` (bottom)

```
arrow-datafusion update-p… ➜ cargo run --example parquet_sql_multiple_files
   Compiling datafusion-examples v36.0.0 (/Users/thauck/personal/code/github.com/tshauck/arrow-datafusion/datafusion-examples)
    Finished dev [unoptimized + debuginfo] target(s) in 1.66s
     Running `target/debug/examples/parquet_sql_multiple_files`
+----+----------+-------------+--------------+---------+------------+-----------+------------+------------------+------------+---------------------+
| id | bool_col | tinyint_col | smallint_col | int_col | bigint_col | float_col | double_col | date_string_col  | string_col | timestamp_col       |
+----+----------+-------------+--------------+---------+------------+-----------+------------+------------------+------------+---------------------+
| 4  | true     | 0           | 0            | 0       | 0          | 0.0       | 0.0        | 30332f30312f3039 | 30         | 2009-03-01T00:00:00 |
+----+----------+-------------+--------------+---------+------------+-----------+------------+------------------+------------+---------------------+
thread 'main' panicked at datafusion-examples/examples/parquet_sql_multiple_files.rs:71:57:
test data path has no parent
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

arrow-datafusion update-p… ➜ cargo run --example parquet_sql_multiple_files
   Compiling datafusion-examples v36.0.0 (/Users/thauck/personal/code/github.com/tshauck/arrow-datafusion/datafusion-examples)
    Finished dev [unoptimized + debuginfo] target(s) in 1.46s
     Running `target/debug/examples/parquet_sql_multiple_files`
+----+----------+-------------+--------------+---------+------------+-----------+------------+------------------+------------+---------------------+
| id | bool_col | tinyint_col | smallint_col | int_col | bigint_col | float_col | double_col | date_string_col  | string_col | timestamp_col       |
+----+----------+-------------+--------------+---------+------------+-----------+------------+------------------+------------+---------------------+
| 4  | true     | 0           | 0            | 0       | 0          | 0.0       | 0.0        | 30332f30312f3039 | 30         | 2009-03-01T00:00:00 |
+----+----------+-------------+--------------+---------+------------+-----------+------------+------------------+------------+---------------------+
Error: "No parent"
```

Reply:
Ah, my apologies. I really should not use the github ui to do reviews as it doesn't show the context properly.